tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -368,24 +369,28 @@ async def with_connector_acceptance_test(context: ConnectorContext, connector_un
         Container: A container with connector acceptance tests installed.
     """
     connector_under_test_image_name = context.connector.acceptance_test_config["connector_image"]
-    await load_image_to_docker_host(context, connector_under_test_image_tar, connector_under_test_image_name)
+    image_sha = await load_image_to_docker_host(context, connector_under_test_image_tar, connector_under_test_image_name)
 
     if context.connector_acceptance_test_image.endswith(":dev"):
         cat_container = context.connector_acceptance_test_source_dir.docker_build()
     else:
         cat_container = context.dagger_client.container().from_(context.connector_acceptance_test_image)
 
-    return (
+    cat_container = (
         with_bound_docker_host(context, cat_container)
         .with_entrypoint([])
         .with_exec(["pip", "install", "pytest-custom_exit_code"])
-        .with_env_variable("CACHEBUSTER", str(uuid.uuid4()))
-        .with_mounted_directory("/test_input", context.get_connector_dir(exclude=["secrets", ".venv"]))
-        .with_directory("/test_input/secrets", context.secrets_dir)
+        .with_mounted_directory("/test_input", context.get_connector_dir())
+    )
+    cat_container = (
+        with_mounted_connector_secrets(context, cat_container, "/test_input/secrets")
+        .with_env_variable("CONNECTOR_IMAGE_ID", image_sha)
```

Comment:
Busts the cache if the connector image under test has changed (image change leads to image sha256 change)